ARCs/arc-0065.md

```diff
@@ -36,23 +36,38 @@ in this document are to be interpreted as described in <a href="https://datatrac
 
 ### In Program Errors
 
-When a program wants to emit a run time error code, contained in the bytecode, 
+When a program wants to emit informative run time errors, contained in the bytecode,
 it **MUST**:
 
-1. Push to the stack the bytes string containing the error code;
+1. Push to the stack the bytes string containing the error;
 1. Execute the `log` opcode to use the bytes from the top of the stack;
 1. Execute the `err` opcode to immediately terminate the program.
 
 Upon a program run time failure, the Algod API returns both the failed *program 
 counter* (`pc`) and the `log` content.
 
+### Error format
+
 > The AVM programs bytecode have limited sized. In this convention, the errors are
-> part of the bytecode. It is **RECOMMENDED** to use fixed length error codes or 
-> short error messages.
+> part of the bytecode, therefore it is good to mind errors' formatting and sizing.
+
+> Errors consist of a _code_ and an optional _short message_.
+
+Errors **MUST** statisfy the regexp: `s/^ERR:(\w*)\s-\s(.*)$/`.
```

Comment:
I realise this regex wouldn't satisfy the below example of `ERR:042` where there isn't a short message.

Maybe the following would be better? Welcome others input here.
`s/^ERR:(\w*)(\s?-\s?(.*))?$/`

Reply:
Just realised my example above created a third capturing group. This may be better: `ERR:(\w*)(?:\s?-\s?(.*))?` Which then gives you 2 capturing groups, the first for the error code, the second for the error message.

Reply:
Good point! Also: both numeric (`ERR:0042`) and alphanumeric error codes should be supported (either "labels" or things like "hex" codes, like `ERR:BAD-REQUEST` or `ERR:0xAA`). I'll update the spec!